src/BuildMetadata.ts

```diff
@@ -21,6 +21,9 @@ export class BuildMetadata {
     private static readonly MINOR_VERSION_MULTIPLIER = 100;
     private static readonly PATTERN = /^([\d]+)\.([\d]+)(?:\.([\d]+))?(-[\w]+)?(-SNAPSHOT)?(-BETA-.)?$/;
 
+    /* tslint:disable:member-ordering */
+    public static readonly VERSION_3_8 = BuildMetadata.MAJOR_VERSION_MULTIPLIER * 3 + BuildMetadata.MINOR_VERSION_MULTIPLIER * 8;
```

Comment:
why is name of variable bound to a specific version?

Reply:
It is needed on near cache implementation. Near cache behaves differently after 'VERSION_3_8'.

Reply:
@furkansenharputlu I think we should not define specific versions in `BuildMetadata` as this class contains information about current specific release. We can introduce `Versions` class as in Java if this is neccessary.

Additionally, I think we should introduce a new method that takes major, minor and patch as parameters and creates the specific number as the current `calculateVersion` does. Current approach to create a specific version number is error prone.

Reply:
Sorry I mixed `BuildMetadata` and `BuildInfo` classes. You can dismiss my first paragraph but the second still applies